2-projects/1-todo-list/script.js

```diff
@@ -13,13 +42,51 @@ let todos = [
 populateTodoList(todos);
 
 // This function will take the value of the input field and add it as a new todo to the bottom of the todo list. These new todos will need the completed and delete buttons adding like normal.
+
 function addNewTodo(event) {
   // The code below prevents the page from refreshing when we click the 'Add Todo' button.
   event.preventDefault();
   // Write your code here... and remember to reset the input field to be blank after creating a todo!
+
+  const inputBar = document.getElementById("todoInput");
+  const todoText = inputBar.value;
+
+  // If no input then it returns to stop creating todos
+
+  if (!todoText) {
+    return;
+  }
+
+  let newToDo = document.createElement("li");
+  newToDo.classList.add("new-todo-item");
+
+  let todoTextSpan = document.createElement("span");
+  todoTextSpan.textContent = todoText;
+  newToDo.appendChild(todoTextSpan);
+
+  let addNewTodoBtn = event.target;
+  addNewTodoBtn.addEventListener("click", addNewTodo);
+
+  let removeButton = document.getElementById("remove-all-completed");
+
+  removeButton.addEventListener("click", () => {
+    newToDo.remove();
+  });
+
+  let todoList = document.getElementById("todo-list");
+  todoList.appendChild(newToDo);
+
+  inputBar.value = "";
 }
 
```

Comment:
What's missing here is adding an event handler to the add todo button that calls our `addNewTodo` function.